/work/nashokkumar_umass_edu/nipschal/Neurips-Challenge-22/predict_graph_embed.py:433: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  labels = torch.tensor(dataset_tensor[:, 1, :].clone().detach(), dtype=torch.long)
PermutedDKT
Number of questions:  1855
Number of students:  6468
Number of concepts: 1076
Using batch size: 16
Using batch size: 16
Successfull in data prepration!
Successfully loaded the optimizer
======== Epoch 1 / 200 ========
Training...
Step: 0
Parameter containing:
tensor([[ 0.0176, -0.0043, -0.0069,  ...,  0.0078, -0.0162, -0.0071],
        [ 0.0013, -0.0143,  0.0060,  ..., -0.0026, -0.0293,  0.0127],
        [ 0.0211, -0.0170,  0.0166,  ...,  0.0090, -0.0208, -0.0270],
        ...,
        [ 0.0150,  0.0151, -0.0164,  ...,  0.0204, -0.0170,  0.0084],
        [-0.0241,  0.0057, -0.0067,  ...,  0.0254,  0.0080, -0.0168],
        [ 0.0233, -0.0070, -0.0070,  ..., -0.0066, -0.0253,  0.0031]],
       device='cuda:0', requires_grad=True)
Traceback (most recent call last):
  File "/work/nashokkumar_umass_edu/nipschal/Neurips-Challenge-22/predict_graph_embed.py", line 490, in <module>
    main()
  File "/work/nashokkumar_umass_edu/nipschal/Neurips-Challenge-22/predict_graph_embed.py", line 475, in main
    model, epoch_train_loss, epoch_val_loss = train(epochs, dkt_model, train_dataloader, val_dataloader, optimizer, scheduler) # add val_dataloader later
  File "/work/nashokkumar_umass_edu/nipschal/Neurips-Challenge-22/predict_graph_embed.py", line 346, in train
    loss, acc = model(b_input_ids, b_labels, epoch_i+1)
  File "/work/nashokkumar_umass_edu/.conda/envs/nistorch/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/work/nashokkumar_umass_edu/nipschal/Neurips-Challenge-22/predict_graph_embed.py", line 235, in forward
    hidden_states, _ = self.gru(input_embed, epoch)
  File "/work/nashokkumar_umass_edu/.conda/envs/nistorch/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/work/nashokkumar_umass_edu/nipschal/Neurips-Challenge-22/predict_graph_embed.py", line 168, in forward
    hidden = self.cell(x, lower, hidden)
  File "/work/nashokkumar_umass_edu/.conda/envs/nistorch/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/work/nashokkumar_umass_edu/nipschal/Neurips-Challenge-22/predict_graph_embed.py", line 62, in forward
    W_hr = self.W_hr * lower
RuntimeError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 47.46 GiB total capacity; 41.28 GiB already allocated; 19.44 MiB free; 46.59 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF